shim/src/handle_vcpu_kvm_get_regs.c

```diff
@@ -36,8 +48,57 @@
  *   @return SHIM_SUCCESS on success, SHIM_FAILURE on failure.
  */
 NODISCARD int64_t
-handle_vcpu_kvm_get_regs(struct kvm_regs *const pmut_ioctl_args) NOEXCEPT
+handle_vcpu_kvm_get_regs(
+    struct shim_vcpu_t const *const pmut_vcpu, struct kvm_regs *const pmut_ioctl_args) NOEXCEPT
 {
-    (void)pmut_ioctl_args;
+
+    struct mv_rdl_t *const pmut_rdl = (struct mv_rdl_t *const)shared_page_for_current_pp();
+    platform_expects(NULL != pmut_rdl);
+    platform_expects(MV_INVALID_HANDLE != g_mut_hndl);
```

Comment:
Move this to the top of the function. The platform_expects(NULL != pmut_rdl); is correct as it is paired with the code that fetches pmut_rdl, so that is stating that the narrow contract is pmut_rdl  cannot be NULL. The function should start with the narrow contract stating that the handle must be valid.